src/App.tsx

```diff
@@ -1,26 +1,244 @@
+
+    return todoServices
+      .updateTodo(updatedTodo)
+      .then(newTodo => {
+        setTodos(currentTodos => {
+          return currentTodos.map(todo => {
+            if (todo.id === newTodo.id) {
+              return newTodo;
+            }
+
+            return todo;
+          });
+        });
+      })
+      .catch(error => {
+        setErrorMessage(ErrorType.ERROR_UPDATE);
+        throw error;
+      })
+      .finally(() => {
+        setLoadingByIds(prev => prev.filter(id => id !== todoToUpdate.id));
+      });
+  };
+
+  const toggleAll = () => {
+    const uncompletedTodos = todos.filter(todo => !todo.completed);
+
+    if (uncompletedTodos.length > 0) {
+      uncompletedTodos.forEach(todo => {
+        updateTodoStatus({ ...todo, completed: true });
+      });
+    } else {
+      todos.forEach(todo => {
+        updateTodoStatus({ ...todo, completed: false });
+      });
+    }
+  };
+
+  const updateTodoTitle = (updatedTodo: Todo) => {
+    setErrorMessage(ErrorType.ERROR_DEFAULT);
+    setLoadingByIds(prev => [...prev, updatedTodo.id]);
+
+    const findTodo = todos.find(todo => todo.id === updatedTodo.id) as Todo;
+
+    if (findTodo?.title.trim() === updatedTodo.title.trim()) {
+      setLoadingByIds([]);
+
+      return;
+    }
+
+    if (updatedTodo.title.trim() === '') {
+      deleteTodo(updatedTodo.id);
+
+      return;
+    }
+
+    return todoServices
+      .updateTodo(updatedTodo)
+      .then(newTodo => {
+        setTodos(currentTodos => {
+          return currentTodos.map(todo => {
+            if (todo.id === newTodo.id) {
+              return newTodo;
+            }
+
+            return todo;
+          });
+        });
+      })
+      .catch(error => {
+        setErrorMessage(ErrorType.ERROR_UPDATE);
+        throw error;
+      })
+      .finally(() => {
+        setLoadingByIds(prev => prev.filter(id => id !== updatedTodo.id));
+      });
+  };
+
+  if (!todoServices.USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">
-          React Todo App - Add and Delete
-        </a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <Header
+          isVisibileBtn={isVisibileBtn}
+          isAllCompletedTodo={isAllCompletedTodo}
+          inputRef={inputRef}
+          isLoading={isLoading}
+          setErrorMessage={setErrorMessage}
+          onSubmit={addTodo}
+          toggleAll={toggleAll}
+        />
+
+        <TodoList
+          filteredTodos={filteredTodos}
+          tempTodo={tempTodo}
+          isLoading={isLoading}
```

Comment:
```suggestion
          isLoading={!!tempTodo}
```
should be the same without isLoading state